core/src/main/scala/fs2/async/mutable/Topic.scala

```diff
@@ -0,0 +1,250 @@
+package fs2.async.mutable
+
+import fs2.Async.Change
+import fs2._
+import fs2.Stream._
+
+/**
+  * Asynchronous Topic.
+  *
+  * Topic allows you to distribute `A` published by arbitrary number of publishers to arbitrary number of subscribers.
+  *
+  * Topic has built-in back-pressure support implemented as maximum bound (`maxQueued`) that subscriber is allowed to enqueue.
+  * Once that bound is hit, the publish may hold on until subscriber consumes some of its published elements .
+  *
+  * Additionally the subscriber has possibility to terminate whenever size of enqueued elements is over certain size
+  * by using `subscribeSize`.
+  *
+  *
+  */
+trait Topic[F[_],A] {
+
+  /**
+    * Published any elements from source of `A` to this topic. This will stop publishing if the source has no elements
```

Comment:
Ok the comment is not that up-to-date. I changed signature to sink. What I want to say here is that this will asynchronously block if any subscriber is over is maxQueued threshold. Does that make a sense?